src/main/java/soconnect/model/person/UniquePersonList.java

```diff
@@ -131,6 +131,29 @@ public void changeRelevantPersonTag(Tag oldTag, Tag newTag) {
         }
     }
 
+    /**
+     * Removes the tag in every contact.
+     *
+     * @param tag The tag to be removed.
+     */
+    public void removeRelevantPersonTag(Tag tag) {
+        for (int i = 0; i < internalList.size(); i++) {
+            Person oldPerson = internalList.get(i);
+            List<Tag> personTags = new ArrayList<>(oldPerson.getTags());
+            if (!personTags.contains(tag)) {
+            } else {
+                personTags.remove(tag);
+                Set<Tag> updatedTags = new HashSet<>(personTags);
+                Person updatedPerson = new Person(oldPerson.getName(),
+                    oldPerson.getPhone(),
+                    oldPerson.getEmail(),
+                    oldPerson.getAddress(),
+                    updatedTags);
+                internalList.set(i, updatedPerson);
```

Comment:
```suggestion
    public void removeRelevantPersonTag(Tag tag) {
        for (int i = 0; i < internalList.size(); i++) {
            Person oldPerson = internalList.get(i);
            List<Tag> personTags = new ArrayList<>(oldPerson.getTags());
            if (personTags.contains(tag)) {
                personTags.remove(tag);
                Set<Tag> updatedTags = new HashSet<>(personTags);
                Person updatedPerson = new Person(oldPerson.getName(),
                    oldPerson.getPhone(),
                    oldPerson.getEmail(),
                    oldPerson.getAddress(),
                    updatedTags);
                internalList.set(i, updatedPerson);
```

Why do you have this method for `Todo` since the tag delete feature is done by Clarence? Also, might have to check with Clarence if this will affect his implementation.

Reply:
When tags are deleted, I have to sync the deleting of tags with the `Todo`s. Hence, I imported some of his code over to create the corresponding methods for `Todo`. The above code is one of them, which I didn't bother to refactor as it will result in a merge conflict.

Reply:
Okay, then this should be `removeRelevantTodoTag(Tag tag)` right since it is removing the deleted tags in `Todo`s.

Then remember to update the tag delete command, make sure it calls this `removeRelevantTodoTag` when a `tag` is deleted.